### PR TITLE
HEEDLS-295 Ignore archived tutorials in section service

### PR DESCRIPTION
**Pull request by benjimarshall**
### Changes
Ignore archived sections in the tutorial list of the section page,
and account for this when looking for the next section. 

This includes changes from review comments in #152.

### Testing
Amend tests to not use incorrect old stored procedure, apply this
change for both the sections, and the next section as well.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs [hunk 1]
```diff
@@ -382,6 +402,62 @@ public void Get_section_content_returns_null_if_customisation_is_inactive()
...
+
+            // When
+            var result = sectionContentService.GetSectionContent(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().NotBeNull();
+            result!.SectionScore.Should().Be(0);
+            result!.MaxSectionScore.Should().Be(2); // Not 3 as uspReturnSectionsForCandCust_V2 returns because
+                                                    // it counts archived tutorial 9366
+            result!.DiagnosticAttempts.Should().Be(1);
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: Is this relevant to the test? 🙂

**Review by FridaTveit — COMMENTED** [review]
One minor comment, otherwise looks good 🙂

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: Which bit? The comment isn't strictly necessary but I thought it was helpful to precisely locate the discrepancy for whoever uses this test in the future 🙂

Comment (FridaTveit) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: Sorry I meant checking the diagnostic attempts. It didn't seem relevant to what was being tested? I like the comment, I agree it's useful 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: So the diagnostic attempts and the section score do both depend on whether archived tutorials are included or not (the section score being the sum of the tutorial scores, and the section diagnostic attempts being the maximum over the diagnostic attempt values for each tutorials in the section). As it happens both of these agree with the old stored procedure for this case, but they might not for every case. I think it's helpful to include because it checks the three values that including archived tutorials can effect, even if there only is a discrepancy in one value 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: Alternatively I could temporarily edit the database to make all three of these values depend on whether we are including the archived tutorials or not? 🙂

Comment (FridaTveit) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:421]: Ah I see. I think changing the db (in transaction scope) so they all depend on archived tutorials would make it a more useful test 🙂 I'd also be tempted to split it into two tests; one for the max score and one for the current score 🙂

Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs]: Yes I agree, added in 187795d 🙂

File: DigitalLearningSolutions.Data.Tests/Helpers/SectionContentTestHelper.cs [hunk 2]
```diff
@@ -15,11 +15,6 @@ public SectionContentTestHelper(SqlConnection connection)
             this.connection = connection;
         }
 
-        public IEnumerable<OldTutorial> TutorialsFromOldStoredProcedure(int progressId, int sectionId)
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Helpers/SectionContentTestHelper.cs:18]: Are there other helper methods like this which we can remove, as I think we removed most if not all of the tests comparing our results with the old stored procedures now? 🙂

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs [hunk 3]
```diff
@@ -415,10 +415,37 @@ public void Get_section_content_should_not_use_archived_tutorials_in_scores()
...
+                // This will should not change the overall DiagnosticAttempts because the tutorial has now been archived
+                sectionContentTestHelper.UpdateDiagAttempts(tutorialToArchive, progressId, 2);
+
+                // When
+                var result = sectionContentService.GetSectionContent(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.SectionScore.Should().Be(7);
+                result!.MaxSectionScore.Should().Be(10);
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs:446]: I don't think we should be testing the max score here, as that's covered by the other test? 🙂

**Review by FridaTveit — COMMENTED** [review]
A couple of comments otherwise looks good 🙂

File: DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs (hunk 3 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs]: Yep! Removed in 0f6b296 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂
